de/model.py

```diff
@@ -4,48 +4,90 @@
 import math
 import numpy as np
 import matplotlib.pyplot as plt
-from ODE import solver
+import matplotlib.colors as colors
+import matplotlib.cm as cmx
+from .ODE import solver, jacobian_autograd
 
 class Model():
     '''
     The class for basic model simulation
     '''
-    def __init__(self, n_x=83, pert=50, index=np.loadtxt('./data/node_Index.csv', delimiter=',', dtype=str), dt=0.1, N_t=100, filename='./figures/simulation_1.jpg'):
+    def __init__(self, n_x=83, N=84, pert=50, index='./data/node_Index.csv', dt=0.1, N_t=48, save_path='./figures/simulation_1.jpg', expdata_path='./data/exp_data.csv'):
         self.n_x = n_x #The number of components involved
+        self.N = N #The number of knockout conditions
         self.x0 = np.ones(n_x) #Initial values
         self.pert = pert # The strength of pertubation
-        self.index = index # The gene name for each component
-        self.dt, self.N_t = dt, N_t # Time series
-        self.filename = filename #Path for saving simulation result
-        self.t, self.sol = self.sim(self) #Time series and simulation of ODE
-        self.eps, self.w, self.alpha, self.beta = self.random_params(self) #Randomly initialized parameters
-
-    def graph(self):
-        '''Function receives simulation of ODE and the number of components involved, then generate the graph'''
+        #TODO: Fix paths for loading data
+        #self.index = np.loadtxt(index, delimiter=',', dtype=str) # The gene name for each component
+        self.dt, self.N_t = dt, N_t # Time step and total length of time for generating time series
+        self.save_path = save_path #Path for saving simulation result
+        self.expdata_path = expdata_path #Path for inputing experiment data
+        self.t = None #Time series
+        self.sol = None #Simulation of ODE
+        self.p = None # Array of eps (value that bounds saturation effect), w (interaction strength), alpha (degradation rate)
+        self.beta = None #Knock-out effects
+        self.sse = None # The sum of square error across all measurements
+        self.jacb = None # The jacobian matrix of the system
+    def graph(self, cond):
+        '''Function receives simulation of ODE and the number of components involved, then generate the graph.
```

Comment:
Try to have one line of space between the functions. Helps with clarity of the code.